shared/handler.go

```diff
@@ -6,14 +6,12 @@ import (
 	"github.com/pokt-network/pocket/shared/types"
 )
 
-// TODO: Only supporting a subset of topics because not all are used.
-func (node *Node) handleEvent(event *types.Event) error {
-	switch event.PocketTopic {
-	case types.ConsensusMessage:
-		node.GetBus().GetConsensusModule().HandleMessage(event.MessageData)
+func (node *Node) handleEvent(event *types.PocketEvent) error {
+	switch event.Topic {
+	case types.PocketTopic_CONSENSUS_MESSAGE_TOPIC:
+		node.GetBus().GetConsensusModule().HandleMessage(event.Data)
 	default:
-		// TODO(discuss): Should we panic here?
-		log.Printf("[WARN] Unsupported event: %s \n", event.PocketTopic)
+		log.Printf("Unsupported event: %s \n", event.Topic)
```

Comment:
We should think about handling undefined behavior, including panic recovery and unexpected behaviors such as these. I am talking in general as P2P also lacks this aspect (_intentionally deferred to be thought about thoroughly, perhaps collectively?_)

Reply:
Adding a top-level TODO at the PR level. Would you want to add it to a future protocol hour and lead the discussion?

Reply:
Good suggestion Shans, will do.

Reply:
Thanks Zanzan